src/Macros/errors.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::{Context, Result};
+
+#[doc(hidden)]
+pub fn _add_info_to_result<T>(res: Result<T>, info: String) -> Result<T> {
+    res.with_context(|| info)
+}
+
+/// Add file and line number information + optional context to an anyhow::Result
+#[macro_export]
+macro_rules! info {
+    ($result:expr) => { info!($result, "Caused an error") };
+    ($result:expr, $($args:expr ),*) => { $crate::Macros::errors::_add_info_to_result($result, format!("{}: {}", concat!(file!(), ":", line!(), ":", column!()) ,format!( $( $args ),* ))) };
+}
```

Comment:
The main problem with this is that we cannot get the line information without macros and if we call the macro in the MyContext impl we will always only reference that one line. It only works when macros call it as per the [documentation](https://doc.rust-lang.org/std/macro.file.html). I don't see any way to access line information with traits.

Furthermore if we were to introduce the trait we would almost always write something like `thing.with_context(|| format!("{}", ...))` i personally think `context!(thing, "{}", ...)` is nicer because we avoid the format! and ||. We can't have as nice syntax with traits; the argument must be a closure and we cannot do operator overloading or variable number of arguments. With macros we allow `context!(thing)`, `context!(thing, "Woopsie!")`, `context!(thing, "Unable to get {}", var)`, which is pretty convenient.

Reply:
Ah yeah, that is an important point i completely missed.